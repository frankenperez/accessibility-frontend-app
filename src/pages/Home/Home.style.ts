import styled from "@emotion/styled";

const StyledHome = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;

  h1 {
    width: 100%;
    font-size: 2rem;
    font-weight: 300;
    text-align: center;
    padding: 1rem;
  }

  .gallery {
    display: flex;
    justify-content: center;
    width: 100%;
    overflow-y: hidden;

    img {
      width: 50%;
      height: auto;
    }
  }
`;

export default StyledHome;
