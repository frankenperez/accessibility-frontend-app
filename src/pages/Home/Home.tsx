// React
import React from "react";

// i18n
import { useTranslation } from "react-i18next";

// Styles
import StyledHome from "./Home.style";

// Images
import img01 from "assets/images/img10029747.webp";
import img02 from "assets/images/img7188938.jpeg";

export default function Home() {
  const { t } = useTranslation("app");

  return (
    <StyledHome>
      <h1>{t("home.title")}</h1>
      <div className="gallery">
        <img src={img01}></img>
        <img src={img02}></img>
      </div>
    </StyledHome>
  );
}
