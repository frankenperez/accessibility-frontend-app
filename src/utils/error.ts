import axios from "axios";

export const errorParser = (error: unknown) => {
  let message: string;
  if (axios.isAxiosError(error)) {
    const errorData = error.response?.data as { reason: string } | string;
    if (typeof errorData === "object" && errorData?.reason) {
      message = errorData?.reason;
    } else {
      message = JSON.stringify(errorData);
    }
  } else if (error instanceof Error) {
    message = error.message;
  } else {
    message = JSON.stringify(error);
  }
  return message;
};
