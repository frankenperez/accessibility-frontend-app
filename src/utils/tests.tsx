// React
import React, { PropsWithChildren } from "react";

// Testing Library
import { render, RenderOptions } from "@testing-library/react";

// Router
import { BrowserRouter } from "react-router-dom";

// Internationalization
import en_BG from "../languages/en-GB.json";
import i18next from "i18next";
import { I18nextProvider } from "react-i18next";

// MUI Theme
import { createTheme, StyledEngineProvider, ThemeProvider } from "@mui/material/styles";
import theme from "styles/theme";
const appTheme = createTheme(theme);

// Internationalization config
i18next.init({
  interpolation: { escapeValue: false }, // React already does escaping
  lng: "en", // Language to use
  resources: {
    en: {
      app: en_BG
    }
  }
});

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  route?: string;
}

const renderWithProviders = (
  ui: React.ReactElement,
  {
    route = "/",
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren): JSX.Element => {
    return (
      <I18nextProvider i18n={i18next}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={appTheme}>
            <BrowserRouter>{children}</BrowserRouter>
          </ThemeProvider>
        </StyledEngineProvider>
      </I18nextProvider>
    );
  };
  window.history.pushState({}, "Test page", route);

  // Return an object with the store and all of RTL's query functions
  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders };
