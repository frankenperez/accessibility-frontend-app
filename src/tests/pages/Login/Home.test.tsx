// React
import React from "react";

// Component
import Home from "pages/Home/Home";

// Test Utils
import { renderWithProviders, runAxe } from "utils/tests";

// Tests
describe("Tests for <Home />", () => {
  it("Renders with default props", () => {
    const { container } = renderWithProviders(<Home />);
    // Creates the snapshot test
    expect(container).toMatchSnapshot();
  });
  it("Renders with no accessibility violations", async () => {
    const { container } = renderWithProviders(<Home />);
    // Check a11y violations
    const results = await runAxe(container);
    expect(results).toHaveNoViolations();
  });
});
