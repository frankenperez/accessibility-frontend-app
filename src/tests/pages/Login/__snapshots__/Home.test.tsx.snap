// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Home /> Renders with default props 1`] = `
<div>
  <div
    class="css-13kl4hw"
  >
    <h1>
      The greater the diversity, the greater the perfection
    </h1>
    <div
      class="gallery"
    >
      <img
        src="img10029747.webp"
      />
      <img
        src="img7188938.jpeg"
      />
    </div>
  </div>
</div>
`;
